LIBVmotor0.11/LIBVmotor0.11.ino

```diff
@@ -1,172 +1,249 @@
-
-  //Read each byte from master
-  TV = Wire.read()*10;
-  TI = Wire.read()*100;
-  IE = Wire.read();
-
-  //Each EEPROM address can store (0-255), make sure the data in range or it will lose the correct value
-  EEPROM.update(10, TV/10);
-  EEPROM.update(11, TI/100);
-  EEPROM.update(12, IE);
-  BPM_Timing();
-  Serial.print("TV:");
-  Serial.print(TV);
-  Serial.print(" | TI:");
-  Serial.print(TI);
-  Serial.print(" | IE:");
-  Serial.println(IE*0.5);
-}
+/* LIBV Project
+ * Website: https://bvmvent.org/
+ * 
+ * #VENTILATOR MOTOR CONTROL#
+ * 
+ * NOTE:
+ * >DO NOT put any thing make the loop slow or interrupted,
+ *  the PiD functin need real time to calculate
+ *  - Don't use delay() in 'Loop'
+ *  - Try not to print alot of data with serial in 'Loop'
+ *  - Don't use 'while()'
+ *
+ * LIBRARY:
+ * - Encoder: https://github.com/PaulStoffregen/Encoder
+ * - simplePID: https://github.com/eTRONICSKH/SimplePID-Arduino-Library
+ * - BTS7960 Driver: https://github.com/eTRONICSKH/BTS7960-Driver-Arduino-Library
+ * - button: https://github.com/eTRONICSKH/SimpleButton-Arduino-Library
+ */
+ 
+#include <EEPROM.h>
+#include <Wire.h>
+#include <Encoder.h>
+#include <simplePID.h>
+#include <BTS7960.h>
+#include <button.h>
+
+#define STATION_I2C_ADD 8 //I2C slave station Address
+
+//VAL: EPPROM Address
+const int TV_ADD= 10, TI_ADD=11, IE_ADD=12;
+
+//P-IN: Hall switch
+#define HALL1_PIN 11
+#define HALL2_PIN 10
+
+//P-IN: Button
+#define BREATH_PIN 12
+
+//P-IN: Encoder
+#define chA 2   //Motor's encoder channel A
+#define chB 3   //Motor's encoder channel B
+
+//P-IN: Analog sensor
+#define PRES1_PIN A2
+#define PRES2_PIN A3
+
+//P-OUT: Motor driver, BTS7960
+#define LPWM 5
+#define RPWM 6
+
+//VAL: Motor's spec
+#define mGear 51.0  //Motor's gear ratio
+#define mPpr 11.0   //Number of tick per one revolution capture by the encoder
+#define mPulse 4.0  //Always 4, the encoder library return 4 in 1 tick from the encoder
+
+//VAL: PID Parameters
+#define Kp 140.0    //Proportional gain
+#define Ki 120.0    //Integral gain
+#define ZERO 0.0    //Zero value
+
+//VAR: PID Variable 
+long t, xt, dt;                   //timer
+long En, xEn;                     //Store the encoder value
+float goPos=0.0, aPos=0.0, dPos;  //Store the Position
+float goSpeed=0.0, aSpeed=0.0;    //Target Speed, current Speed
+int cmd;                          //digital command from PID, the PWM value
+
+//Setting Variable
+#define RPB 0.62           //Number of round to rotate to one breath inhale
+int TV, TI, IE;
+float TV_MIN=0, TV_MAX=680;
+float TV2RPB=0.0;
+//bool pushIn=false;                  //define action of push IN or release the bag
+int timeIn, timeOut, timeRest;                //time to push in and push out in millisecond
+float speedIn, speedOut, speedMov;  //speed to push in and push out of the motor, in round per second (rps)
+unsigned long timer=0;              //to store the timer of running, millis(), millisecond
+
+//VAR: Ven stat
+int venStat=0;
```

Comment:
Is stat statistic, state, or status? ;) 

Maybe change this to currentState instead of venStat.